target/linux/ramips/dts/mt7621_xiaomi_mi-router-cr6606.dts

```diff
@@ -0,0 +1,187 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+
+#include "mt7621.dtsi"
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+
+/ {
+	compatible = "xiaomi,mi-router-cr6606", "mediatek,mt7621-soc";
+	model = "Xiaomi Mi Router CR6606";
+
+	aliases {
+		led-boot = &led_sys_yellow;
+		led-failsafe = &led_sys_yellow;
+		led-running = &led_sys_blue;
+		led-upgrade = &led_sys_yellow;
+		label-mac-device = &gmac0;
+	};
+
+	chosen {
+		bootargs = "console=ttyS0,115200n8";
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		led_sys_yellow: sys_yellow {
+			label = "yellow:sys";
+			gpios = <&gpio 14 GPIO_ACTIVE_LOW>;
+		};
+
+		led_sys_blue: sys_blue {
+			label = "blue:sys";
+			gpios = <&gpio 16 GPIO_ACTIVE_LOW>;
+		};
+
+		net_yellow {
+			label = "yellow:net";
+			gpios = <&gpio 13 GPIO_ACTIVE_LOW>;
+		};
+
+		net_blue {
+			label = "blue:net";
+			gpios = <&gpio 15 GPIO_ACTIVE_LOW>;
+		};
+	};
+
+	keys {
+		compatible = "gpio-keys";
+
+		reset {
+			label = "reset";
+			gpios = <&gpio 18 GPIO_ACTIVE_LOW>;
+			debounce-interval = <60>;
+			linux,code = <KEY_RESTART>;
+		};
+
+		wps {
+			label = "wps";
+			gpios = <&gpio 7 GPIO_ACTIVE_LOW>;
+			debounce-interval = <60>;
+			linux,code = <KEY_WPS_BUTTON>;
+		};
+	};
+};
+
+&nand {
+	status = "okay";
+
+	partitions {
+		compatible = "fixed-partitions";
+		#address-cells = <1>;
+		#size-cells = <1>;
+
+		partition@0 {
+			label = "Bootloader";
+			reg = <0x0 0x80000>;
+			read-only;
+		};
+
+		partition@80000 {
+			label = "Nvram";
+			reg = <0x80000 0x40000>;
+			read-only;
+		};
+
+		partition@c0000 {
+			label = "Bdata";
+			reg = <0xc0000 0x40000>;
+			read-only;
+		};
+
+		factory: partition@100000 {
+			label = "Factory";
+			reg = <0x100000 0x80000>;
+			read-only;
+		};
+
+		partition@180000 {
+			label = "crash";
+			reg = <0x180000 0x40000>;
+			read-only;
+		};
+
+		partition@1c0000 {
+			label = "crash_log";
+			reg = <0x1c0000 0x40000>;
+			read-only;
+		};
+
+		partition@200000 {
```

Comment:
This should be a kernel partition, not a partition for the entire firmware. The label should be 'kernel' and the compatible string should be removed.

Reply:
Actually, CR660x series factory firmware layout looks exactly the same. For OpenWRT though, it do worth a try.
By the way, is the "kernel-ubi" layout good for FIT image? The current legacy image is a result of an unbootable OpenWrt FIT image, maybe it will change later one day.

Reply:
Overhauled the partition table, squashed most partitions into one "firmware". No recovery tool available so I don't see the need of keeping a stock "firmware", and 30MB for kernel seems like a waste.

Reply:
However, it doesn't seem right here. Generally, `compatible = "denx, uimage"` is not used on nand devices.
```
define Device/xiaomi_mi-router-cr6606
  ...
  KERNEL_SIZE := 4096k
  IMAGE_SIZE := 128512k
```
Since the size of the kernel partition is limited here, the partition table of dts may should be changed to this:
```
partition@200000 {
	label = "kernel";
	reg = <0x200000 0x400000>;
};

partition@600000 {
	label = "ubi";
	reg = <0x600000 0x7980000>;
};
```

Reply:
I see. Take Netis WF2881's approach seems to be a solution.

Reply:
Fixed.